### DISCUSSION TITLE
Different bar color

### DISCUSSION THREAD

**Discussion by elkrien** [qtile/qtile#3385] [Q&A]
Hi!
Is it possible to have different bar color when workspace is empty and when there is any application open?
I would like to have semi-transparent bar background when workspace is empty and full color bar when anything on this workspace is open. Any help how to do it?

**Comment by elParaguayo** [comment]
Of course it's possible!

You should write a function that's linked to the `current_screen_change` hook. Something like this:

``` python
@hook.subscribe.current_screen_change
def change_bar_colour()

    for screen in qtile.screens:
        # Check if screen is currently focused
        if screen == qtile.current_screen:
            screen.top.background = "000000"
        else:
            screen.top.background = "00000060"

        screen.top.draw()
```

Untested and you'd need to tailor it for your bar's positions etc. but hopefully you get the idea.

**Reply by elParaguayo** [reply]
Oops. totally misread your question! Hang on...

**Reply by elParaguayo** [reply]
Similar sort of approach needed and you may need multiple hooks (but you also probably don't need all the ones I've listed!)

``` python
@hook.subscribe.changegroup
@hook.subscribe.client_focus
@hook.subscribe.client_managed
@hook.subscribe.client_killed
@hook.subscribe.client_new
@hook.subscribe.group_window_add
def change_bar_colour(*args)

    for screen in qtile.screens:
        # Check if group has applications open
        if screen.group.windows:
            screen.top.background = "000000"
        else:
            screen.top.background = "00000060"

        screen.top.draw()
